### PR TITLE
Add severity configuration e2e tests

### LINKED ISSUE #10385: [Test]: Add E2E test of the BuildCheck with applied severity
### Issue Description

After the `Severity` updates were merged https://github.com/dotnet/msbuild/pull/10330 , the e2e tests of BuildCheck were skipped in CI. 
Let's add automated E2E test of severity set to all possible values after the PR: https://github.com/dotnet/msbuild/pull/10307 enabling tests back is merged.  

### Goal
To automate testing around editorconfig in e2e scenarios. 
Currently if the severity was not applied correctly for some reason in e2e there is no verification test for that

### PR DESCRIPTION
**Pull request by f-alizada**
Fixes #10385 

### Context
E2E tests to cover the configurability and the effect of the severity setup. 

### Changes Made
Introduced 2 additional tests:
- Severity != `None` should result the BuildCheck BC0101 should report with proper severity.
- Severity = `None` should result the BuildCheck BC0101 not to report anything.

### Testing
The PR itself is the introduction of tests

### CODE REVIEW

File: src/BuildCheck.UnitTests/EndToEndTests.cs [hunk 1]
```diff
@@ -105,6 +105,36 @@ public void SampleAnalyzerIntegrationTest_ReplayBinaryLogOfAnalyzedBuild(bool bu
...
+    [InlineData("error", "error BC0101")]
+    [InlineData("suggestion", "BC0101")]
+    [InlineData("default", "warning BC0101")]
+    public void EditorConfig_SeverityAppliedCorrectly(string BC0101Severity, string expectedOutputValues)
+    {
+        PrepareSampleProjectsAndConfig(true, out TransientTestFile projectFile, BC0101Severity);
+
+        string output = RunnerUtilities.ExecBootstrapedMSBuild(
+            $"{Path.GetFileName(projectFile.Path)} /m:1 -nr:False -restore -analyze",
+            out bool success, false, _env.Output, timeoutMilliseconds: 120_000);
```
Comment (JanKrivanek) [inline src/BuildCheck.UnitTests/EndToEndTests.cs:119]: Not necessarily to be aprt of this PR - we should remove the timeout explicit parameter (here and in the other tests) and rely on sane default of 30 seconds.
2 minutes seems excessive here. I know it is from the other tests - but that was added as a temporary desperate measure for the randomly staling integration test - which is now resolved.

File: src/BuildCheck.UnitTests/EndToEndTests.cs [hunk 2]
```diff
@@ -105,6 +105,36 @@ public void SampleAnalyzerIntegrationTest_ReplayBinaryLogOfAnalyzedBuild(bool bu
         }
     }
 
+    [Theory]
+    [InlineData("warning", "warning BC0101")]
+    [InlineData("error", "error BC0101")]
+    [InlineData("suggestion", "BC0101")]
```
Comment (JanKrivanek) [inline src/BuildCheck.UnitTests/EndToEndTests.cs:111]: is there a way to distinguish suggestion from the others?
If not by presence of specific text, than maybe via absence of specific texts (e.g. `{ "warning BC0101", "error BC0101" }`)?

File: src/BuildCheck.UnitTests/EndToEndTests.cs (hunk 1 above)
Comment (JanKrivanek) [reply src/BuildCheck.UnitTests/EndToEndTests.cs:119]: If we'd add `string[] unexpectedOutputValues` argument to the parametrized test - than we'd be able to more reliably test the `suggestion` case, as well as include the `none` case here and avoid duplication.

**Review by JanKrivanek — APPROVED** [review]
Looks good. I added couple thoughts for consideration

File: src/BuildCheck.UnitTests/EndToEndTests.cs (hunk 1 above)
Comment (f-alizada) [reply src/BuildCheck.UnitTests/EndToEndTests.cs:119]: I was thinking to do some refactoring there in this PR, but decided to do that after this is merged, overall agree. 

Thank you for the suggestion `unexpectedOutputValues` working on that

File: src/BuildCheck.UnitTests/EndToEndTests.cs (hunk 2 above)
Comment (f-alizada) [reply src/BuildCheck.UnitTests/EndToEndTests.cs]: Thank you for the suggestion, update the test case 
Duplication removed :)
